addons/modules/functions/fnc_moduleRemoveArsenal.sqf

```diff
@@ -0,0 +1,35 @@
+#include "script_component.hpp"
+/*
+ * Author: CreepPork, Kex
+ * Zeus module function to remove the virtual arsenal.
+ *
+ * Arguments:
+ * 0: Logic <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [LOGIC] call zen_modules_fnc_moduleRemoveArsenal
+ *
+ * Public: No
+ */
+
+params ["_logic"];
+
+private _object = attachedTo _logic;
+deleteVehicle _logic;
+
+if (isNull _object) exitWith {
+    [LSTRING(NoObjectSelected)] call EFUNC(common,showMessage);
+};
+
+if (EGVAR(common,preferredArsenal) == 1 && {isClass (configFile >> "CfgPatches" >> "ace_arsenal")}) then {
+    [_object, true] call ace_arsenal_fnc_removeBox;
+} else {
+    // Set variables manually, since "AmmoboxExit" is not reliable
+    _object setVariable ["BIS_addVirtualWeaponCargo_cargo", nil, true];
+    private _boxes = missionNamespace getVariable ["BIS_fnc_arsenal_boxes", []];
+    _boxes = _boxes - [_object];
```

Comment:
Nitpick but don't need the local variable, can just do the `_boxes - [_object]` in the `setVariable` below.

Reply:
Well, I could sure make an ugly one-liner just to get rid of the local variable `_boxes`, but why though?

Reply:
Not about shortening code, more-so that I thought the code would look cleaner as:
```sqf
private _boxes = missionNamespace getVariable ["BIS_fnc_arsenal_boxes", []];
missionNamespace setVariable ["BIS_fnc_arsenal_boxes", _boxes - [_object], true];
```